zuko/transforms.py

```diff
@@ -661,34 +656,27 @@ def __init__(
         self.slope = self._bernstein_poly(x, dtheta, dbasis)
 
     @staticmethod
-    def _constrain_theta(unconstrained_theta: Tensor, smooth_bounds: bool, bound: float) -> Tensor:
-        r"""Processes the unconstrained output of the hyper-network to be increasing."""
-
-        if smooth_bounds:
-            unconstrained_theta = torch.cat(
-                (
-                    unconstrained_theta[..., :1],
-                    unconstrained_theta,
-                    unconstrained_theta[..., -1:],
-                ),
-                dim=-1,
-            )
-
-        if bound:
-            theta_min = -bound * torch.ones_like(unconstrained_theta[..., :1])
-
-            def fn(x):
-                return torch.nn.functional.softmax(x, dim=-1) * 2 * bound
-        else:
-            shift = math.log(2.0) * unconstrained_theta.shape[-1] / 2
-            theta_min = unconstrained_theta[..., :1] - shift
-            unconstrained_theta = unconstrained_theta[..., 1:]
-            fn = torch.nn.functional.softplus
+    def _constrain_theta(unconstrained_theta: Tensor) -> Tensor:
+        """Processes the unconstrained output of the hyper-network to be increasing."""
+        shift = math.log(2.0) * unconstrained_theta.shape[-1] / 2
+
+        theta_min = unconstrained_theta[..., :1]
+        unconstrained_theta = unconstrained_theta[..., 1:]
+
+        # ensure smooth bounds
```

Comment:
Did you derive the math of these yourself or is it based on some reference? If the latter it might be worth linking to it. Same for the bounded transform actually.

Reply:
I think this was initially @oduerr's idea.

He observed, that the small initial values for thetas lead to a rather sub optimal initialization and hence decided to add this offset to distribute the symmetrically around the normal base distribution.

$\log(2)$ is used because $\text{softplus}(0) = \log(2)$